h5py/_hl/files.py

```diff
@@ -35,8 +35,13 @@
     swmr_support = True
 
 
-libver_dict = {'earliest': h5f.LIBVER_EARLIEST, 'latest': h5f.LIBVER_LATEST}
+libver_dict = {'earliest': h5f.LIBVER_EARLIEST}
+if hdf5_version >= (1, 10, 2):
+    libver_dict.update({'v108': h5f.LIBVER_V18, 'v110': h5f.LIBVER_V110})
 libver_dict_r = dict((y, x) for x, y in six.iteritems(libver_dict))
+libver_dict.update({'latest': h5f.LIBVER_LATEST})
+if hdf5_version < (1, 10, 2):
+    libver_dict_r = dict((y, x) for x, y in six.iteritems(libver_dict))
```

Comment:
I've just spotted that this recreation should be unnecessary; just finish defining `libver_dict` before creating `libver_dict_r`, and then it only needs to be done once.

Reply:
Ah, no, I understand. If `LIBVER_LATEST` is an alias for `LIBVER_V110`, then we'd be relying on dict ordering, which is only consistent from Python 3.6. That's tricky. We might want a comment explaining why it's done like that.

Reply:
I've rewritten it in what I think is a clearer way, at the cost of a bit of repetition.